packages/web/src/components/spa/HackAUBG/RegistrationForm/RegistrationForm.jsx

```diff
@@ -0,0 +1,732 @@
+import React, { useEffect } from 'react'; // eslint-disable-line
+import './registration_form.css';
+import { useForm } from 'react-hook-form';
+import { useState } from 'react';
+import axios from 'axios';
+import { url } from '../../../../Global';
+import { BsExclamationCircleFill } from 'react-icons/bs';
+
+const RegistrationForm = () => {
+    const {
+        register,
+        handleSubmit,
+        formState: { errors } // eslint-disable-line
+    } = useForm({ mode: 'all' });
+
+    const [loadingAnimation, setLoadingAnimation] = useState(false);
+    const [submitPressed, setSubmitPressed] = useState(false);
+    const [submitButtonValue, setSubmitButtonValue] = useState('Register'); // eslint-disable-line
+    const [apiError, setApiError] = useState(false);
+    const [buttonMessage, setButtonMessage] = useState('');
+    const [disableTeamNameField, setDisableTeamNameField] = useState(true);
+
+    // ? What are these used for?
+    // eslint-disable-next-line no-unused-vars
+
+    const registerMember = (data) => {
+        axios({
+            method: 'post',
+            // TODO: Remove the url param when done with testing the endpoint
+            // * this disables the mailing feature -> e.g. it won't send emails to random emails you use to test the registration with
+            url: url + '/api/hackathon/register?testing=true',
+            data
+        })
+            // eslint-disable-next-line no-unused-vars
+            .then((res) => {
+                setLoadingAnimation(false);
+                setApiError(false); // remove red color of button
+            })
+            .catch((err) => {
+                setLoadingAnimation(false);
+                getErrorMessage(err);
+                setApiError(true); // put button in error state
+                setSubmitButtonValue('Retry');
+            });
+    };
+
+    function parseVars(data) {
+        Object.entries(data).forEach((field) => {
+            if (field[1] === 'True' || field[1] === 'False') {
+                data[field[0]] = field[1] === 'True' ? true : false;
+            }
+        });
+
+        return data;
+    }
+    function getErrorMessage(err) {
+        if (err['response']['status'] >= 500) {
+            setButtonMessage('Something went wrong');
```

Comment:
Good, I like it. Consider renaming the function tho, probably something like `setErrorMessage`
Also you should inform the user that they should contact us if they keep experiencing a "Something went error" issue.